include/z64save.h

```diff
@@ -252,6 +252,16 @@ typedef enum {
     /* 3 */ GAMEMODE_END_CREDITS
 } GameMode;
 
+typedef enum {
+    /* 0 */ SCENE_LAYER_CHILD_DAY,
+    /* 1 */ SCENE_LAYER_CHILD_NIGHT,
+    /* 2 */ SCENE_LAYER_ADULT_DAY,
+    /* 3 */ SCENE_LAYER_ADULT_NIGHT    
+} SceneLayer;
```

Comment:
```suggestion
typedef enum {
    /* 0 */ SCENE_LAYER_CHILD_DAY,
    /* 1 */ SCENE_LAYER_CHILD_NIGHT,
    /* 2 */ SCENE_LAYER_ADULT_DAY,
    /* 3 */ SCENE_LAYER_ADULT_NIGHT
    /* 4 */ SCENE_LAYER_CUTSCENE_0,
    /* 5 */ SCENE_LAYER_CUTSCENE_1,
    /* 6 */ SCENE_LAYER_CUTSCENE_2,
    /* 7 */ SCENE_LAYER_CUTSCENE_3,
    /* 8 */ SCENE_LAYER_CUTSCENE_4,
    /* 9 */ SCENE_LAYER_CUTSCENE_5,
    /* A */ SCENE_LAYER_CUTSCENE_6,
    /* B */ SCENE_LAYER_CUTSCENE_7,
    /* C */ SCENE_LAYER_CUTSCENE_8,
} SceneLayer;

#define SCENE_LAYER_HYRULE_FIELD_CHILD 0
#define SCENE_LAYER_HYRULE_FIELD_CHILD_OCARINA_OF_TIME 1
#define SCENE_LAYER_KOKIRI_FOREST_ADULT_HOSTILE 2
#define SCENE_LAYER_KOKIRI_FOREST_ADULT_PEACEFUL 3
```
It seems wrong to introduce this enumeration but then still use raw values anyway.

Additionally day/night, child/adult is true for most cases, but there is a special case for Hyrule Field and Kokiri Forest where this pattern is overridden. 

In Hyrule Field, 0 and 1 chosen depending on whether the Ocarina of Time can be obtained, where 0 is the "normal" scene setup for Child, and 1 is loaded whenever the player is to obtain the Ocarina of Time. 

In Kokiri Forest, 2 and 3 are chosen depending on whether the Forest Temple is completed, with 2 being the forest with enemies, and 3 being the forest with everyone safe and the deku sprout spawned.

Reply:
The cutscene layer enums will come from the individual scenes. General purpose defines for each cutscene dont make much sense, when you can have scene specific ones that can be named more precisely for their purpose.

I am aware of the special cases where the pattern is overridden, that is why they do not use this enum currently. They will use the hyrule field and kokiri forest enums respectively, when they are made

Reply:
take for example, kakariko
an enum like such would live in the spot01 header (whever they actually get committed), and can be used whenever the layer is being checked in kakariko specific contexts
```
typedef enum {
    /* 4 */ SPOT01_LAYER_NOCTURNE_PART1 = SCENE_LAYER_CUTSCENE_START
    /* 5 */ SPOT01_LAYER_NOCTURNE_PART2
    /* 6 */ SPOT01_LAYER_WELL_DRAINING
    ...
} KakarikoSceneLayer;
```

Reply:
would you also have scene specific cutscene indexes?

Reply:
I am almost thinking that cutscene indexes should be in terms of layers, something like `GET_CS_INDEX(SPOT01_LAYER_WELL_DRAINING)`

but i havent fleshed out the idea and determined if it makes sense yet